parsing/extend.ml

```diff
@@ -79,8 +79,10 @@ type ('a,'b,'c) ty_user_symbol =
 
 (** {5 Type-safe grammar extension} *)
 
-type norec = NoRec    (* just two *)
-type mayrec = MayRec  (* incompatible types *)
+(* Should be merged with gramlib's implementation *)
+
+type norec = Gramlib.Grammar.ty_norec
+type mayrec = Gramlib.Grammar.ty_mayrec
```

Comment:
Note that an alternative way would be to declare the Gramlib phantom types as open variants, it's a common design pattern. Those types are structural, so that we could declare the same here without ever referencing Gramlib.